internal/cli/status.go

```diff
@@ -146,6 +156,458 @@ func (c *StatusCommand) Run(args []string) int {
 	return 0
 }
 
+// RefreshApplicationStatus takes a project and application target and generates
+// a list of applications to refresh the status on. If all projects are requested
+// to be refreshed, the CLI will do its best to honor the request. However if
+// a project is local and the CLI was not invoked inside that project dir, the
+// CLI won't be able to refresh that projects application statuses.
+func (c *StatusCommand) RefreshApplicationStatus(projectTarget, appTarget string) error {
+	// Get our API client
+	client := c.project.Client()
+
+	// Get the entire list of apps
+	// Determine project locality
+	// Do the Work (local or remote)
+	if c.flagAllProjects {
+		resp, err := client.ListProjects(c.Ctx, &empty.Empty{})
+		if err != nil {
+			c.ui.Output("Failed to retrieve all projects:"+clierrors.Humanize(err), terminal.WithErrorStyle())
+			return err
+		}
+		allProjectsRef := resp.Projects
+
+		for _, pRef := range allProjectsRef {
+			projectResp, err := client.GetProject(c.Ctx, &pb.GetProjectRequest{
+				Project: pRef,
+			})
+			if err != nil {
+				return err
+			}
+			project := projectResp.Project
+
+			workspace, err := c.getWorkspaceFromProject(projectResp)
+			if err != nil {
+				return err
+			}
+
+			// local or remote project?
+			var remote bool
+			if project.DataSource != nil {
+				switch project.DataSource.Source.(type) {
+				case *pb.Job_DataSource_Local:
+					remote = false
+				case *pb.Job_DataSource_Git:
+					remote = true
+				default:
+					remote = false
+				}
+			}
+
+			if !remote {
+				c.ui.Output("Cannot refresh every projects status without command being "+
```

Comment:
```suggestion
				c.ui.Output("Cannot refresh every project's status without command being "+
```

Reply:
I'm not able to hit this error. I have a project dir that is not a git repo, navigate one layer up so I'm not inside the project dir, run `waypoint status -refresh ex2`, then get stuck on `Refreshing status for app "ex2-app" in project "ex2". Refreshing could take a while...`
A sigkill gives me this https://gist.github.com/krantzinator/b899a5b7d7f4cb5e39f6286c4e54bfc6

Reply:
You'll only hit this error if you run `waypoint status -refresh -all-projects`. It should fail pretty quickly though for a local project so there's something to investigate and fix here it sounds like

Reply:
Well, we axed refreshing statuses on all projects so this is no longer seen. I've also reworked how app statuses are refreshed, so you might not hit this error. If you do let me know!